monitoring/definitions/shared/container.go

```diff
@@ -87,3 +87,45 @@ var (
 		}
 	}
 )
+
+type ContainerMonitoringGroupOptions struct {
+	// CPUUsage transforms the default observable used to construct the CPU usage panel.
+	CPUUsage func(observable Observable) Observable
+
+	// MemoryUsage transforms the default observable used to construct the memory usage panel.
+	MemoryUsage func(observable Observable) Observable
+
+	// IOUsage transforms the default observable used to construct the IO usage panel.
+	IOUsage func(observable Observable) Observable
+}
+
+// NewContainerMonitoringGroup creates a group containing panels displaying
+// container monitoring metrics - cpu, memory, io resource usage as well as
+// a container missing alert - for the given container.
+func NewContainerMonitoringGroup(containerName string, owner monitoring.ObservableOwner, options *ContainerMonitoringGroupOptions) monitoring.Group {
+	if options == nil {
+		options = &ContainerMonitoringGroupOptions{}
+	}
+	if options.CPUUsage == nil {
+		options.CPUUsage = NoopObservableTransformer
+	}
+	if options.MemoryUsage == nil {
+		options.MemoryUsage = NoopObservableTransformer
+	}
+	if options.IOUsage == nil {
+		options.IOUsage = NoopObservableTransformer
+	}
+
+	return monitoring.Group{
+		Title:  TitleContainerMonitoring,
+		Hidden: true,
+		Rows: []monitoring.Row{
+			{
+				ContainerMissing(containerName, owner).Observable(),
+				options.CPUUsage(ContainerCPUUsage(containerName, owner)).Observable(),
```

Comment:
WDYT about an API like:

```suggestion
				applyOptionalOption(options.CPUUsage, ContainerCPUUsage(containerName, owner)),
```

where:

```go
func applyOptionalOption(opt ObservableOption, o sharedObservable): Observable {
    if opt == nil {
      return o.Observable(),
   } 
   return opt(o).Observable(),
}
```

then we can drop `NoopObservableTransformer` and all the `if opt.foo == nil { ... }` code blocks!

Reply:
another thought: make `func(observable Observable) Observable` a `ObservableOption`, then:

```go
func (o *ObservableOption) maybeApply(...) Observable {
   if (o == nil) { ... }
   ...
}
```

Reply:
Updated.